site/static/api/device-registry-revised.yaml

```diff
@@ -0,0 +1,713 @@
+               $ref: '#/components/responses/NotAllowed'
+            404:
+               $ref: '#/components/responses/NotFound'
+            412:
+               $ref: '#/components/responses/ResourceVersionMismatch'
+
+      delete:
+         tags:
+            - tenants
+         summary: Delete tenant
+         operationId: deleteTenant
+         parameters:
+            - $ref: '#/components/parameters/tenant'
+            - $ref: '#/components/parameters/resourceVersion'
+         responses:
+            204:
+               description: object deleted
+            401:
+               $ref: '#/components/responses/Unauthorized'
+            403:
+               $ref: '#/components/responses/NotAllowed'
+            404:
+               $ref: '#/components/responses/NotFound'
+            412:
+               $ref: '#/components/responses/ResourceVersionMismatch'
+
+# Device API
+
+   /devices/{tenant}:
+      post:
+         tags:
+            - devices
+         summary: Create new device registration with auto-generated ID
+         operationId: createDeviceRegistration
+         parameters:
+            - $ref: '#/components/parameters/tenant'
+         requestBody:
+            description: New device
+            content:
+               application/json:
+                  schema:
+                     $ref: '#/components/schemas/Device'
+            required: true
+         responses:
+            201:
+               $ref: '#/components/responses/Created'
+            400:
+               $ref: '#/components/responses/MalformedRequest'
+            401:
+               $ref: '#/components/responses/Unauthorized'
+            403:
+               $ref: '#/components/responses/NotAllowed'
+
+   /devices/{tenant}/{deviceId}:
+      post:
+         tags:
+            - devices
+         summary: Create new device registration
+         operationId: createDeviceRegistrationWithId
+         parameters:
+            - $ref: '#/components/parameters/tenant'
+            - $ref: '#/components/parameters/deviceId'
+         requestBody:
+            description: New device
+            content:
+               application/json:
+                  schema:
+                     $ref: '#/components/schemas/Device'
+            required: true
+         responses:
+            201:
+               description: device registered
+               headers:
+                  Location:
+                     description: URL to the resource
+                     schema:
+                        type: string
+                        format: uri
+                  ETag:
+                     description: Version of the resource
+                     schema:
+                        type: string
+            400:
+               $ref: '#/components/responses/MalformedRequest'
+            401:
+               $ref: '#/components/responses/Unauthorized'
+            403:
+               $ref: '#/components/responses/NotAllowed'
+            409:
```

Comment:
if no read access it should answer 404 as well.. Having 403 to mask conflict and 404 to mask 403 allows someone to poke around and find outs existing devices ID, if i am correct.